tools/analyze_logs.py

```diff
@@ -0,0 +1,151 @@
+            slowest_epoch + 1, ave_time_over_epoch[slowest_epoch]))
+        print('fastest epoch {:02d}, average time is {:.4f}'.format(
+            fastest_epoch + 1, ave_time_over_epoch[fastest_epoch]))
+        print('time std over epochs is {:.4f}'.format(std_over_epoch))
+        print('average iter time: {:.4f} s/iter'.format(np.mean(all_times)))
+        print()
+
+
+def plot_curve(log_dicts, args):
+    # if legend is None, use file name of json logs as legend
+    legend = args.legend if args.legend is not None else args.json_logs
+    assert len(legend) == len(args.json_logs)
+    metric = args.key
+
+    for i, log_dict in enumerate(log_dicts):
+        print('plot curve of {}, metric is {}'.format(args.json_logs[i],
+                                                      metric))
+        epochs = log_dict.keys()
+        assert metric in log_dict[list(epochs)
+                                  [0]], '{} does not contain metric {}'.format(
+                                      args.json_logs[i], metric)
+
+        if metric in ('bbox_mAP', 'segm_mAP'):
+            xs = np.arange(1, max(epochs) + 1)
+            ys = []
+            for epoch in epochs:
+                ys += log_dict[epoch][metric]
+            ax = plt.gca()
+            ax.set_xticks(xs)
+            plt.title('{} v.s. epoch'.format(metric))
+            plt.xlabel('epoch')
+            plt.ylabel(metric)
+            plt.plot(xs, ys, label=legend[i], marker='o')
+        elif metric in ('loss', ):
+            xs = []
+            ys = []
+            num_iters_per_epoch = log_dict[list(epochs)[0]]['iter'][-1]
+            for epoch in epochs:
+                iters = log_dict[epoch]['iter']
+                if log_dict[epoch]['mode'][-1] == 'val':
+                    iters = iters[:-1]
+                xs.append(np.array(iters) + (epoch - 1) * num_iters_per_epoch)
+                ys.append(np.array(log_dict[epoch]['loss']))
+            xs = np.concatenate(xs)
+            ys = np.concatenate(ys)
+            plt.title('{} v.s. iter'.format(metric))
+            plt.xlabel('iter')
+            plt.ylabel(metric)
+            plt.plot(xs, ys, label=legend[i], linewidth=0.5)
+        plt.legend()
+    out = args.out
+    if out is None:
+        if not osp.exists('analysis_log_res'):
+            os.mkdir('analysis_log_res')
+        out = 'analysis_log_res/{}.pdf'.format(metric)
+    print('save curve to: {}'.format(out))
+    plt.savefig(out)
+    plt.cla()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Analyze Json Log')
+    parser.add_argument(
+        'task',
+        type=str,
+        choices=['plot_curve', 'cal_train_time'],
+        help='currently support plot curve and calculate average train time')
+    parser.add_argument(
+        'json_logs',
+        type=str,
+        nargs='+',
+        help='path of train log in json format')
+    parser.add_argument(
+        '--key',
```

Comment:
Make this a list, e.g., `--key bbox_mAP segm_mAP`, whick will plot two curves in the same figure.

Reply:
keys are not restricted in ['bbox_mAP', 'segm_mAP', 'loss'], it can be any items in the log, e.g., `loss_cls`.